packages/date-picker/src/shared/components/DateInput/DateInputSegment/DateInputSegment.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { ChangeEventHandler, KeyboardEventHandler } from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useForwardedRef } from '@leafygreen-ui/hooks';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { keyMap, rollover } from '@leafygreen-ui/lib';
+import { Size } from '@leafygreen-ui/tokens';
+import { useUpdatedBaseFontSize } from '@leafygreen-ui/typography';
+
+import {
+  charsPerSegment,
+  defaultMax,
+  defaultMin,
+  defaultPlaceholder,
+} from '../../../constants';
+import { getSegmentMaxLength, getValueFormatter } from '../../../utils';
+import { useDatePickerContext } from '../../DatePickerContext';
+
+import {
+  baseStyles,
+  fontSizeStyles,
+  segmentSizeStyles,
+  segmentThemeStyles,
+  segmentWidthStyles,
+} from './DateInputSegment.styles';
+import { DateInputSegmentProps } from './DateInputSegment.types';
+
+/**
+ * Renders a single date segment with the
+ * appropriate character padding/truncation.
+ *
+ * Only fires a change handler when the input is blurred
+ */
+export const DateInputSegment = React.forwardRef<
+  HTMLInputElement,
+  DateInputSegmentProps
+>(
+  (
+    {
+      segment,
+      value,
+      min: minProp,
+      max: maxProp,
+      onChange,
+      onBlur,
+      onKeyDown,
+      ...rest
+    }: DateInputSegmentProps,
+    fwdRef,
+  ) => {
+    const min = minProp ?? defaultMin[segment];
+    const max = maxProp ?? defaultMax[segment];
+
+    const inputRef = useForwardedRef(fwdRef, null);
+
+    const { theme } = useDarkMode();
+    const baseFontSize = useUpdatedBaseFontSize();
+    const { size, disabled } = useDatePickerContext();
+    const formatter = getValueFormatter(segment);
+    const pattern = `[0-9]{${charsPerSegment.year}}`;
+    const maxLength = getSegmentMaxLength(segment);
+
+    /** Prevent non-numeric values from triggering a change event */
+    const handleChange: ChangeEventHandler<HTMLInputElement> = e => {
+      const { target } = e;
+      const numericValue = Number(target.value);
+
+      if (!isNaN(numericValue)) {
+        const stringValue = numericValue.toString();
+        onChange?.({
```

Comment:
should this onChange be a required prop?

Reply:
It _could_ be a required prop into `DateInputSegment`, since it will always be provided by `DateInputBox`.

I left it as optional, since `onChange` is not a required prop on `DatePicker` itself